### ISSUE TITLE
Is there any documentation regarding Android tools are to be used with net7/8-android?

### ISSUE THREAD

**Issue by beeradmoore** [xamarin/xamarin-android#8315]
### Android application type

.NET Android (net7.0-android, etc.)

### Affected platform version

VS4Mac 17.6.3

### Description

TL;DR; is there any regularly updated documentation that defines the specific Android tooling (command-line version, build-tools version, platform-tools version) to use with net7/8-android?

Longer;
With the [announcement](https://devblogs.microsoft.com/visualstudio/visual-studio-for-mac-retirement-announcement/) that VS4Mac is going away I started looking as to how to manage my Android SDK and build tools as I've always used VS4Mac to manage my Android SDKs and various build tools. VS4Mac (and Windows AFAIK) has an `Approved List (Recommended)` and a `Full List (Unsupported)` of the various Android tooling it can install. [This](https://github.com/xamarin/xamarin-android/blob/c31dcae4dbad45bd7c1f3b84f0792b0b5840c842/Documentation/guides/building-apps/build-properties.md#androidmanifesttype) seems to indicate that as of Xamarin.Android 13.0 we should no longer worry about the `Approved List`?

Regarding installing Android SDK it appears that the MAUI [documentation](https://learn.microsoft.com/en-au/dotnet/maui/get-started/installation?tabs=visual-studio-code#android-setup) lists 4 ways to install Android SDK, with the first and recommended way being `dotnet build -t:InstallAndroidDependencies target`. 

This appears to be mentioned in the Xamarin-android repo a few times (eg, [here](https://github.com/xamarin/xamarin-android/blob/c31dcae4dbad45bd7c1f3b84f0792b0b5840c842/Documentation/guides/building-apps/build-targets.md#installandroiddependencies), [here](https://github.com/xamarin/xamarin-android/blob/c31dcae4dbad45bd7c1f3b84f0792b0b5840c842/Documentation/guides/building-apps/build-properties.md#androidmanifesttype)) but with no clear information on what it installs and to what versions (at least from what I can understand). Is this listed anywhere in the repo so we can keep an eye on it and update our systems as things change? Or is the general idea now just to go latest SDK/platform/build tools that there are?

Side note: I can't see any of the above when looking through the [Android docs](https://learn.microsoft.com/en-us/xamarin/android/), is this info only available while browsing this repo? I see some [commits](https://github.com/xamarin/xamarin-android/commit/79be326349d638584321d008e4cf51af3aa97eb9) about syncing with MicrosoftDocs but I am not entirely sure if the intent was for these markdown files be on the full Android docs or not.

Jumping back to the [MAUI docs](https://learn.microsoft.com/en-au/dotnet/maui/get-started/installation?tabs=visual-studio-code#using-installandroiddependencies) it gives a list in the troubleshooting of minimum requirements. 

> Check your minimum versions of installed Android dependencies:
> build-tools >= 32.0.0
> cmdline-tools == 7.0
> platforms;android-33*
> .NET 7: platform-tools = 33.0.2
> .NET 8: platform-tools = 34.0.1

Is the the same versions that will be installed by `dotnet build -t:InstallAndroidDependencies target`? Similar to above, is this listed anywhere so people using Android Studio can match version numbers with its SDK manager?

Does `InstallAndroidDependencies` just install the minimum requirements? Eg. the above lists build-tools >= 32.0.0. Will it install 32.0.0, or will it go for the latest approved  33.0.2, or because we are using Xamarin.Android 13 will it just go to the latest which is 34.0.1? If latest, will it skip if 32.0.0 is found to already be installed?  

Some other general questions which is related to this topic,
Is there any specific reason why cmdline-tools is v7.0 and hasn't been bumped? Like can I expect problems if I roll with v11.0 and it just hasn't been bumped because team are all busy bees over there.

When `InstallAndroidDependencies` is run I believe it gets to this at some point,
```
<CalculateProjectDependencies
    AndroidApiLevel="$(_AndroidApiLevel)"
    TargetFrameworkVersion="$(TargetFrameworkVersion)"
    CommandLineToolsVersion="$(AndroidCommandLineToolsVersion)"
    ManifestFile="$(_ProjectAndroidManifest)"
    BuildToolsVersion="$(AndroidSdkBuildToolsVersion)"
    PlatformToolsVersion="$(AndroidSdkPlatformToolsVersion)"
    NdkVersion="$(AndroidNdkVersion)"
    NdkRequired="$(_NdkRequired)"
  >
```
I _think_ we can set `AndroidCommandLineToolsVersion`, `AndroidSdkBuildToolsVersion` and `AndroidSdkPlatformToolsVersion` in our csproj. Is there any reason why not to do this so devs and CI/CD systems are all using the same versions to build things?

Additional context:
This was a result of discussions in and around [this Twitter thread](https://twitter.com/beeradmoore/status/1697150220772225056) which only ended up having more questions than answers, so I figured I'd come to the source to ask. Huge thanks for Gerald and Bret for helping out with doc links.

### Steps to Reproduce

N/A

**Comment by jonathanpeppers** [comment]
The versions for the `InstallAndroidDependencies` target mostly comes from here:

https://github.com/xamarin/xamarin-android-tools/blob/9c50a2d08038d91bbf3dd66c250dd480623a60d4/src/Xamarin.Android.Tools.AndroidSdk/Xamarin.Android.Tools.Versions.props#L12-L17

There are also relevant docs pages, such as:

* https://learn.microsoft.com/xamarin/android/deploy-test/building-apps/build-targets#installandroiddependencies
* https://learn.microsoft.com/xamarin/android/deploy-test/building-apps/build-properties#androidsdkbuildtoolsversion

These are the ones we sync from this repo to MS docs. We can always improve them if you find something completely missing.

> Is there any reason why not to do this so devs and CI/CD systems are all using the same versions to build things?

This is probably only needed if you want to go the extra mile for 100% reproduceable builds. Honestly, I would go with what is preinstalled on AzDO build agents -- and only think about pinning/installing an older version if something goes wrong. We try to make all the versions of the Android SDK work. Sometimes (as with any third-party dependency) new versions come out and break things. For local development, the Android SDK that stable Android Studio provides *should work* most of the time as well.

I'll let @jonpryor chime in if he has other comments, though. 😄

**Comment by beeradmoore** [comment]
Thanks for coming through with the goods @jonathanpeppers! This makes a lot of sense.

> We can always improve them if you find something completely missing

I think the problem was I went to docs, went to getting started and setting installing SDKs but couldn't find any info on `InstallAndroidDependencies`. Whereas the docs I was after are located in `Deployment and Testing > Building Apps > Build Targets`. It makes sense why the docs are there as it isn't install tool instructions but part of the build system that happens to be able to also install build tools.

Using the search and looking for "InstallAndroidDependencies" will find it, but if you do what I did and search for "install SDK" you end up with a whole bunch of pages which are not what I am after (or even very helpful).

RE: 
> Is there any specific reason why cmdline-tools is v7.0 and hasn't been bumped?
I thought the difference of v7 to v11 meant we are 4 years behind (like how a new Android SDK comes out every year). But the latest v7 came out mid last year. 

Happy to close this and get it off your plate.

**Comment by jonathanpeppers** [comment]
> But the latest v7 came out mid last year.

I think the cadence is the team that works on the Xamarin/MAUI tooling in Visual Studio & the installer, they try to bump all the Android SDK components about once per VS release like 17.7, 17.8, etc. So, things could be slightly behind occasionally.

If customers start using Android SDKs from Android Studio that update more often -- we might have to rethink that strategy. Thanks! I think we can close for now? Feel free to post other questions if you have them.
